src/rational.rs

```diff
@@ -25,19 +25,46 @@ impl ToSql<Binary, Sqlite> for Rational {
     }
 }
 
+#[derive(Debug)]
+struct InvalidBlob;
+
+impl std::fmt::Display for InvalidBlob {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "Invalid blob data in sqlite")
+    }
+}
+
+impl std::error::Error for InvalidBlob {}
+
+// Not (yet?) in the standard library.
+// From https://internals.rust-lang.org/t/slice-split-at-should-have-an-option-variant/17891
+#[inline]
+fn get_split_at(slice: &[u8], mid: usize) -> Option<(&[u8], &[u8])> {
+    if mid > slice.len() {
+        None
+    } else {
+        Some(slice.split_at(mid))
+    }
+}
+
 impl FromSql<Binary, Sqlite> for Rational {
     fn from_sql(bytes: SqliteValue<'_, '_, '_>) -> deserialize::Result<Self> {
         let bytes = <*const [u8] as FromSql<Binary, Sqlite>>::from_sql(bytes)?;
         let bytes: &[u8] = unsafe { &*bytes };
 
-        let numer_len = u32::from_le_bytes(bytes[0..4].try_into().unwrap());
+        let (header, values) = get_split_at(bytes, 4).ok_or(InvalidBlob {})?;
+        let numer_len = u32::from_le_bytes(header.try_into().unwrap()) as usize;
 
-        let (numer, denom) = bytes[4..].split_at(numer_len as usize);
+        let (numer, denom) = get_split_at(values, numer_len).ok_or(InvalidBlob {})?;
 
-        Ok(Rational(Ratio::new(
-            BigUint::from_bytes_le(numer),
-            BigUint::from_bytes_le(denom),
-        )))
+        let numer = BigUint::from_bytes_le(numer);
+        let denom = BigUint::from_bytes_le(denom);
+
+        if denom.is_zero() {
+            return Err(Box::new(InvalidBlob {}));
```

Comment:
```suggestion
            return Err(Box::new(InvalidBlob));
```

Reply:
Or even this 🤔 

```suggestion
            return Err(InvalidBlob.into());
```

I guess I like the symmetry with the `?`-sites, where the error is implicitly converted to the target type.